### PR TITLE
Add micronaut-json-schema feature.

### LINKED ISSUE #2431: Features for JSON Schema
The feature should add the following dependencies: 

```
annotationProcessor("io.micronaut.json-schema:micronaut-json-schema-processor")
implementation("io.micronaut.json-schema:micronaut-json-schema-annotations")
```

Application class should be annotated with: 

```java
@JsonSchemaConfiguration(
    baseUri = "http://localhost:8080/schemas"
)
```

### PR DESCRIPTION
**Pull request by wetted**
closes #2431

This needs to target 4.5.x once that branch is created

### CODE REVIEW

**Comment by wetted** [comment]
ping @sdelamo Should this support all application types?

**Comment by sdelamo** [comment]
> ping @sdelamo Should this support all application types?

Yes.

**Review by sdelamo — CHANGES_REQUESTED** [review]
Remove `JsonSchemaConfiguration`. I think we may change the base uri to `http.//localhost:8080/schemas`. 

Create an extra feature to add[ json-schema-validation](https://github.com/micronaut-projects/micronaut-json-schema/pull/19) module to the test class path. 

Add static-resources configuration to expose the features.

**Comment by wetted** [comment]
> Create an extra feature to add[ json-schema-validation](https://github.com/micronaut-projects/micronaut-json-schema/pull/19) module to the test class path.

@sdelamo What kind of logic do you want for adding one and/or the other, or validating the presence of them?
- [x] add `json-schema-validator` feature if not present when selecting `json-schema` feature?
- [x] `FeatureValidator` to assert `json-schema` is present if `json-schema-validator` is selected?
- [ ]  something else?

Edit: I did what's checked. Can make changes if desired.

File: starter-core/src/main/java/io/micronaut/starter/feature/json/JsonSchemaFeature.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        return Category.API;
+    }
+
+    @Override
+    public void apply(GeneratorContext generatorContext) {
+        generatorContext.addDependency(JSON_SCHEMA_PROCESSOR_DEPENDENCY);
+        generatorContext.addDependency(JSON_SCHEMA_ANNOTAIONS_DEPENDENCY);
+
+        ApplicationConfiguration configuration = generatorContext.getConfiguration();
+        configuration.put("micronaut.router.static-resources.jsonschema.paths", "classpath:META-INF/schemas");
```
Comment (sdelamo) [inline starter-core/src/main/java/io/micronaut/starter/feature/json/JsonSchemaFeature.java]: Please, use `ContributingStaticResources` api.

File: starter-core/src/main/java/io/micronaut/starter/feature/json/JsonSchemaValidationFeature.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import io.micronaut.starter.application.ApplicationType;
+import io.micronaut.starter.application.generator.GeneratorContext;
+import io.micronaut.starter.build.dependencies.Dependency;
+import io.micronaut.starter.build.dependencies.MicronautDependencyUtils;
+import io.micronaut.starter.feature.Category;
+import io.micronaut.starter.feature.Feature;
+import jakarta.inject.Singleton;
+
+@Singleton
+public class JsonSchemaValidationFeature implements Feature {
```
Comment (sdelamo) [inline starter-core/src/main/java/io/micronaut/starter/feature/json/JsonSchemaValidationFeature.java:28]: Apply JsonSchemaFeature if `json-schema-validation` is selected.

File: starter-core/src/main/java/io/micronaut/starter/feature/json/JsonSchemaValidationFeatureValidator.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import io.micronaut.starter.application.ApplicationType;
+import io.micronaut.starter.feature.Feature;
+import io.micronaut.starter.feature.validation.FeatureValidator;
+import io.micronaut.starter.options.Options;
+import jakarta.inject.Singleton;
+
+import java.util.Set;
+
+@Singleton
+public class JsonSchemaValidationFeatureValidator implements FeatureValidator {
```
Comment (sdelamo) [inline starter-core/src/main/java/io/micronaut/starter/feature/json/JsonSchemaValidationFeatureValidator.java]: delete

**Review by sdelamo — CHANGES_REQUESTED** [review]

**Comment by wetted** [comment]
CI is failing on `:test-features:test` because `ThymeleafSpec` generates a project with `views-fieldset-tck`, and the TCK test fails. I'll see if I can figure it out.

Edit: see #2435

**Comment by wetted** [comment]
ping @sdelamo I merged the `ThymeleafSpec` fix and CI is green now. This should be GTG.

**Comment by sdelamo** [comment]
> ping @sdelamo I merged the `ThymeleafSpec` fix and CI is green now. This should be GTG.

Let's wait until we have a release of Micronaut JSON Schema.

**Pull request merged**